File: bindings/lua/Polycode/MaterialManager.lua
class "MaterialManager"


function MaterialManager:__getvar(name)
	if name == "premultiplyAlphaOnLoad" then
		return Polycode.MaterialManager_get_premultiplyAlphaOnLoad(self.__ptr)
	elseif name == "clampDefault" then
		return Polycode.MaterialManager_get_clampDefault(self.__ptr)
	elseif name == "mipmapsDefault" then
		return Polycode.MaterialManager_get_mipmapsDefault(self.__ptr)
	elseif name == "keepTextureData" then
		return Polycode.MaterialManager_get_keepTextureData(self.__ptr)
	end
end

function MaterialManager:__setvar(name,value)
	if name == "premultiplyAlphaOnLoad" then
		Polycode.MaterialManager_set_premultiplyAlphaOnLoad(self.__ptr, value)
		return true
	elseif name == "clampDefault" then
		Polycode.MaterialManager_set_clampDefault(self.__ptr, value)
		return true
	elseif name == "mipmapsDefault" then
		Polycode.MaterialManager_set_mipmapsDefault(self.__ptr, value)
		return true
	elseif name == "keepTextureData" then
		Polycode.MaterialManager_set_keepTextureData(self.__ptr, value)
		return true
	end
	return false
end
function MaterialManager:MaterialManager(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.MaterialManager(unpack(arg))
	end
end

function MaterialManager:Update(elapsed)
	local retVal = Polycode.MaterialManager_Update(self.__ptr, elapsed)
end

function MaterialManager:createTexture(width, height, imageData, clamp, createMipmaps, type)
	local retVal = Polycode.MaterialManager_createTexture(self.__ptr, width, height, imageData.__ptr, clamp, createMipmaps, type)
	if retVal == nil then return nil end
	local __c = _G["Texture"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createNewTexture(width, height, clamp, createMipmaps, type)
	local retVal = Polycode.MaterialManager_createNewTexture(self.__ptr, width, height, clamp, createMipmaps, type)
	if retVal == nil then return nil end
	local __c = _G["Texture"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createTextureFromImage(image, clamp, createMipmaps)
	local retVal = Polycode.MaterialManager_createTextureFromImage(self.__ptr, image.__ptr, clamp, createMipmaps)
	if retVal == nil then return nil end
	local __c = _G["Texture"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createTextureFromFile(fileName, clamp, createMipmaps, resourcePool)
	local retVal = Polycode.MaterialManager_createTextureFromFile(self.__ptr, fileName, clamp, createMipmaps, resourcePool.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Texture"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:deleteTexture(texture)
	local retVal = Polycode.MaterialManager_deleteTexture(self.__ptr, texture.__ptr)
end

function MaterialManager:reloadTextures()
	local retVal =  Polycode.MaterialManager_reloadTextures(self.__ptr)
end

function MaterialManager:reloadProgramsAndTextures()
	local retVal =  Polycode.MaterialManager_reloadProgramsAndTextures(self.__ptr)
end

function MaterialManager:reloadPrograms()
	local retVal =  Polycode.MaterialManager_reloadPrograms(self.__ptr)
end

function MaterialManager:getTextureByResourcePath(resourcePath)
	local retVal = Polycode.MaterialManager_getTextureByResourcePath(self.__ptr, resourcePath)
	if retVal == nil then return nil end
	local __c = _G["Texture"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createProgramFromFile(programPath)
	local retVal = Polycode.MaterialManager_createProgramFromFile(self.__ptr, programPath)
	if retVal == nil then return nil end
	local __c = _G["ShaderProgram"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:loadMaterialLibraryIntoPool(pool, materialFile)
	local retVal = Polycode.MaterialManager_loadMaterialLibraryIntoPool(self.__ptr, pool.__ptr, materialFile)
end

function MaterialManager:cubemapFromXMLNode(node)
	local retVal = Polycode.MaterialManager_cubemapFromXMLNode(self.__ptr, node.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Cubemap"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:materialFromXMLNode(resourcePool, node)
	local retVal = Polycode.MaterialManager_materialFromXMLNode(self.__ptr, resourcePool.__ptr, node.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Material"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createMaterial(resourcePool, materialName, shaderName)
	local retVal = Polycode.MaterialManager_createMaterial(self.__ptr, resourcePool.__ptr, materialName, shaderName)
	if retVal == nil then return nil end
	local __c = _G["Material"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:setShaderFromXMLNode(resourcePool, node)
	local retVal = Polycode.MaterialManager_setShaderFromXMLNode(self.__ptr, resourcePool.__ptr, node.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Shader"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createShaderFromXMLNode(resourcePool, node)
	local retVal = Polycode.MaterialManager_createShaderFromXMLNode(self.__ptr, resourcePool.__ptr, node.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Shader"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:createShader(resourcePool, shaderType, name, vpName, fpName, screenShader)
	local retVal = Polycode.MaterialManager_createShader(self.__ptr, resourcePool.__ptr, shaderType, name, vpName, fpName, screenShader)
	if retVal == nil then return nil end
	local __c = _G["Shader"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:loadMaterialsFromFile(resourcePool, fileName)
	local retVal = Polycode.MaterialManager_loadMaterialsFromFile(self.__ptr, resourcePool.__ptr, fileName)
	if retVal == nil then return nil end
	for i=1,count(retVal) do
		local __c  = _G["Material"]("__skip_ptr__")
		__c.__ptr = retVal[i]
		retVal[i] = __c
	end
	return retVal
end

function MaterialManager:loadShadersFromFile(resourcePool, fileName)
	local retVal = Polycode.MaterialManager_loadShadersFromFile(self.__ptr, resourcePool.__ptr, fileName)
	if retVal == nil then return nil end
	for i=1,count(retVal) do
		local __c  = _G["Shader"]("__skip_ptr__")
		__c.__ptr = retVal[i]
		retVal[i] = __c
	end
	return retVal
end

function MaterialManager:loadCubemapsFromFile(fileName)
	local retVal = Polycode.MaterialManager_loadCubemapsFromFile(self.__ptr, fileName)
	if retVal == nil then return nil end
	for i=1,count(retVal) do
		local __c  = _G["Cubemap"]("__skip_ptr__")
		__c.__ptr = retVal[i]
		retVal[i] = __c
	end
	return retVal
end

function MaterialManager:setAnisotropyAmount(anisotropy)
	local retVal = Polycode.MaterialManager_setAnisotropyAmount(self.__ptr, anisotropy)
end

function MaterialManager:setTextureFilteringMode(textureFilteringMode)
	local retVal = Polycode.MaterialManager_setTextureFilteringMode(self.__ptr, textureFilteringMode)
end

function MaterialManager:getTextureFilteringMode()
	local retVal =  Polycode.MaterialManager_getTextureFilteringMode(self.__ptr)
	return retVal
end

function MaterialManager:addMaterial(material)
	local retVal = Polycode.MaterialManager_addMaterial(self.__ptr, material.__ptr)
end

function MaterialManager:addShader(shader)
	local retVal = Polycode.MaterialManager_addShader(self.__ptr, shader.__ptr)
end

function MaterialManager:getNumShaders()
	local retVal =  Polycode.MaterialManager_getNumShaders(self.__ptr)
	return retVal
end

function MaterialManager:getShaderByIndex(index)
	local retVal = Polycode.MaterialManager_getShaderByIndex(self.__ptr, index)
	if retVal == nil then return nil end
	local __c = _G["Shader"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function MaterialManager:__delete()
	if self then Polycode.delete_MaterialManager(self.__ptr) end
end

File: bindings/javascript/Polycode/MaterialManager.js
function MaterialManager() {
	Object.defineProperties(this, {
		'premultiplyAlphaOnLoad': { enumerable: true, configurable: true, get: MaterialManager.prototype.__get_premultiplyAlphaOnLoad, set: MaterialManager.prototype.__set_premultiplyAlphaOnLoad},
		'clampDefault': { enumerable: true, configurable: true, get: MaterialManager.prototype.__get_clampDefault, set: MaterialManager.prototype.__set_clampDefault},
		'mipmapsDefault': { enumerable: true, configurable: true, get: MaterialManager.prototype.__get_mipmapsDefault, set: MaterialManager.prototype.__set_mipmapsDefault},
		'keepTextureData': { enumerable: true, configurable: true, get: MaterialManager.prototype.__get_keepTextureData, set: MaterialManager.prototype.__set_keepTextureData}
	})
}
MaterialManager.prototype.__get_premultiplyAlphaOnLoad = function() {
	return Polycode.MaterialManager__get_premultiplyAlphaOnLoad(this.__ptr)
}

MaterialManager.prototype.__set_premultiplyAlphaOnLoad = function(val) {
	Polycode.MaterialManager__set_premultiplyAlphaOnLoad(this.__ptr, val)
}

MaterialManager.prototype.__get_clampDefault = function() {
	return Polycode.MaterialManager__get_clampDefault(this.__ptr)
}

MaterialManager.prototype.__set_clampDefault = function(val) {
	Polycode.MaterialManager__set_clampDefault(this.__ptr, val)
}

MaterialManager.prototype.__get_mipmapsDefault = function() {
	return Polycode.MaterialManager__get_mipmapsDefault(this.__ptr)
}

MaterialManager.prototype.__set_mipmapsDefault = function(val) {
	Polycode.MaterialManager__set_mipmapsDefault(this.__ptr, val)
}

MaterialManager.prototype.__get_keepTextureData = function() {
	return Polycode.MaterialManager__get_keepTextureData(this.__ptr)
}

MaterialManager.prototype.__set_keepTextureData = function(val) {
	Polycode.MaterialManager__set_keepTextureData(this.__ptr, val)
}


MaterialManager.prototype.Update = function(elapsed) {
	Polycode.MaterialManager_Update(this.__ptr, elapsed)
}

MaterialManager.prototype.createTexture = function(width,height,imageData,clamp,createMipmaps,type) {
	var retVal = new Texture()
	retVal.__ptr = Polycode.MaterialManager_createTexture(this.__ptr, width,height,imageData,clamp,createMipmaps,type)
	return retVal
}

MaterialManager.prototype.createNewTexture = function(width,height,clamp,createMipmaps,type) {
	var retVal = new Texture()
	retVal.__ptr = Polycode.MaterialManager_createNewTexture(this.__ptr, width,height,clamp,createMipmaps,type)
	return retVal
}

MaterialManager.prototype.createTextureFromImage = function(image,clamp,createMipmaps) {
	var retVal = new Texture()
	retVal.__ptr = Polycode.MaterialManager_createTextureFromImage(this.__ptr, image,clamp,createMipmaps)
	return retVal
}

MaterialManager.prototype.createTextureFromFile = function(fileName,clamp,createMipmaps,resourcePool) {
	var retVal = new Texture()
	retVal.__ptr = Polycode.MaterialManager_createTextureFromFile(this.__ptr, fileName,clamp,createMipmaps,resourcePool)
	return retVal
}

MaterialManager.prototype.deleteTexture = function(texture) {
	Polycode.MaterialManager_deleteTexture(this.__ptr, texture)
}

MaterialManager.prototype.reloadTextures = function() {
	Polycode.MaterialManager_reloadTextures(this.__ptr)
}

MaterialManager.prototype.reloadProgramsAndTextures = function() {
	Polycode.MaterialManager_reloadProgramsAndTextures(this.__ptr)
}

MaterialManager.prototype.reloadPrograms = function() {
	Polycode.MaterialManager_reloadPrograms(this.__ptr)
}

MaterialManager.prototype.getTextureByResourcePath = function(resourcePath) {
	var retVal = new Texture()
	retVal.__ptr = Polycode.MaterialManager_getTextureByResourcePath(this.__ptr, resourcePath)
	return retVal
}

MaterialManager.prototype.createProgramFromFile = function(programPath) {
	var retVal = new ShaderProgram()
	retVal.__ptr = Polycode.MaterialManager_createProgramFromFile(this.__ptr, programPath)
	return retVal
}

MaterialManager.prototype.loadMaterialLibraryIntoPool = function(pool,materialFile) {
	Polycode.MaterialManager_loadMaterialLibraryIntoPool(this.__ptr, pool,materialFile)
}

MaterialManager.prototype.cubemapFromXMLNode = function(node) {
	var retVal = new Cubemap()
	retVal.__ptr = Polycode.MaterialManager_cubemapFromXMLNode(this.__ptr, node)
	return retVal
}

MaterialManager.prototype.materialFromXMLNode = function(resourcePool,node) {
	var retVal = new Material()
	retVal.__ptr = Polycode.MaterialManager_materialFromXMLNode(this.__ptr, resourcePool,node)
	return retVal
}

MaterialManager.prototype.createMaterial = function(resourcePool,materialName,shaderName) {
	var retVal = new Material()
	retVal.__ptr = Polycode.MaterialManager_createMaterial(this.__ptr, resourcePool,materialName,shaderName)
	return retVal
}

MaterialManager.prototype.setShaderFromXMLNode = function(resourcePool,node) {
	var retVal = new Shader()
	retVal.__ptr = Polycode.MaterialManager_setShaderFromXMLNode(this.__ptr, resourcePool,node)
	return retVal
}

MaterialManager.prototype.createShaderFromXMLNode = function(resourcePool,node) {
	var retVal = new Shader()
	retVal.__ptr = Polycode.MaterialManager_createShaderFromXMLNode(this.__ptr, resourcePool,node)
	return retVal
}

MaterialManager.prototype.createShader = function(resourcePool,shaderType,name,vpName,fpName,screenShader) {
	var retVal = new Shader()
	retVal.__ptr = Polycode.MaterialManager_createShader(this.__ptr, resourcePool,shaderType,name,vpName,fpName,screenShader)
	return retVal
}

MaterialManager.prototype.loadMaterialsFromFile = function(resourcePool,fileName) {
	Polycode.MaterialManager_loadMaterialsFromFile(this.__ptr, resourcePool,fileName)
}

MaterialManager.prototype.loadShadersFromFile = function(resourcePool,fileName) {
	Polycode.MaterialManager_loadShadersFromFile(this.__ptr, resourcePool,fileName)
}

MaterialManager.prototype.loadCubemapsFromFile = function(fileName) {
	Polycode.MaterialManager_loadCubemapsFromFile(this.__ptr, fileName)
}

MaterialManager.prototype.setAnisotropyAmount = function(anisotropy) {
	Polycode.MaterialManager_setAnisotropyAmount(this.__ptr, anisotropy)
}

MaterialManager.prototype.setTextureFilteringMode = function(textureFilteringMode) {
	Polycode.MaterialManager_setTextureFilteringMode(this.__ptr, textureFilteringMode)
}

MaterialManager.prototype.getTextureFilteringMode = function() {
	return Polycode.MaterialManager_getTextureFilteringMode(this.__ptr)
}

MaterialManager.prototype.addMaterial = function(material) {
	Polycode.MaterialManager_addMaterial(this.__ptr, material)
}

MaterialManager.prototype.addShader = function(shader) {
	Polycode.MaterialManager_addShader(this.__ptr, shader)
}

MaterialManager.prototype.getNumShaders = function() {
	return Polycode.MaterialManager_getNumShaders(this.__ptr)
}

MaterialManager.prototype.getShaderByIndex = function(index) {
	var retVal = new Shader()
	retVal.__ptr = Polycode.MaterialManager_getShaderByIndex(this.__ptr, index)
	return retVal
}
